File: Tirth Joshi/app/src/main/java/com/example/android_programming_task/MainActivity.java
package com.example.android_programming_task;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    EditText UserName;

    public static final String EXTRA_TEXT = "com.example.android_programming_task.EXTRA_TEXT";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        UserName = findViewById(R.id.editTextTextUsername);


    }

    public void generate(View view) {
        String username = UserName.getText().toString();

        Intent intent = new Intent(this,repo_summary.class);
        intent.putExtra(EXTRA_TEXT,username);
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/androidchallenge/ui/MainActivity.kt
package com.example.androidchallenge.ui

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import com.example.androidchallenge.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        binding = ActivityMainBinding.inflate(layoutInflater)
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        binding.findButton.setOnClickListener {
            Log.d("main", binding.userName.text.toString().trim().replace(' ', '-').lowercase())
            if (binding.userName.text.toString().trim() != "") {
                val intent = Intent(this, RepositoryList::class.java)
                intent.putExtra("userName", binding.userName.text.toString().trim().replace(' ', '-').lowercase())
                startActivity(intent)
                finish()
            } else  {
                Toast.makeText(this, "Enter the user name", Toast.LENGTH_SHORT).show()
            }

        }
    }
}